pkgs/development/compilers/llvm/common/llvm/default.nix

```diff
@@ -48,670 +45,805 @@ let
+        (fetchpatch {
+          name = "signals-gcc-13.patch";
+          url = "https://github.com/llvm/llvm-project/commit/ff1681ddb303223973653f7f5f3f3435b48a1983.patch";
+          hash = "sha256-CXwYxQezTq5vdmc8Yn88BUAEly6YZ5VEIA6X3y5NNOs=";
+          stripLen = 1;
+        })
+        (fetchpatch {
+          name = "base64-gcc-13.patch";
+          url = "https://github.com/llvm/llvm-project/commit/5e9be93566f39ee6cecd579401e453eccfbe81e5.patch";
+          hash = "sha256-PAwrVrvffPd7tphpwCkYiz+67szPRzRB2TXBvKfzQ7U=";
+          stripLen = 1;
+        })
+      ]
+      ++
+        lib.optionals
+          (
+            (lib.versionAtLeast (lib.versions.major release_version) "14")
+            && (lib.versionOlder (lib.versions.major release_version) "17")
+          )
+          [
+            # fix RuntimeDyld usage on aarch64-linux (e.g. python312Packages.numba tests)
+            # See also: https://github.com/numba/numba/issues/9109
+            (fetchpatch {
+              url = "https://github.com/llvm/llvm-project/commit/2e1b838a889f9793d4bcd5dbfe10db9796b77143.patch";
+              relative = "llvm";
+              hash = "sha256-Ot45P/iwaR4hkcM3xtLwfryQNgHI6pv6ADjv98tgdZA=";
+            })
+          ]
+      ++
+        lib.optional (lib.versions.major release_version == "17")
+          # resolves https://github.com/llvm/llvm-project/issues/75168
+          (
+            fetchpatch {
+              name = "fix-fzero-call-used-regs.patch";
+              url = "https://github.com/llvm/llvm-project/commit/f800c1f3b207e7bcdc8b4c7192928d9a078242a0.patch";
+              stripLen = 1;
+              hash = "sha256-e8YKrMy2rGcSJGC6er2V66cOnAnI+u1/yImkvsRsmg8=";
+            }
+          )
+      ++ lib.optionals (lib.versions.major release_version == "18") [
+        # Reorgs one test so the next patch applies
+        (fetchpatch {
+          name = "osabi-test-reorg.patch";
+          url = "https://github.com/llvm/llvm-project/commit/06cecdc60ec9ebfdd4d8cdb2586d201272bdf6bd.patch";
+          stripLen = 1;
+          hash = "sha256-s9GZTNgzLS511Pzh6Wb1hEV68lxhmLWXjlybHBDMhvM=";
+        })
+        # Sets the OSABI for OpenBSD, needed for an LLD patch for OpenBSD.
+        # https://github.com/llvm/llvm-project/pull/98553
+        (fetchpatch {
+          name = "mc-set-openbsd-osabi.patch";
+          url = "https://github.com/llvm/llvm-project/commit/b64c1de714c50bec7493530446ebf5e540d5f96a.patch";
+          stripLen = 1;
+          hash = "sha256-fqw5gTSEOGs3kAguR4tINFG7Xja1RAje+q67HJt2nGg=";
+        })
+      ]
+      ++
+        lib.optionals (lib.versionAtLeast release_version "17" && lib.versionOlder release_version "19")
+          [
+            # Fixes test-suite on glibc 2.40 (https://github.com/llvm/llvm-project/pull/100804)
+            (fetchpatch {
+              url = "https://github.com/llvm/llvm-project/commit/1e8df9e85a1ff213e5868bd822877695f27504ad.patch";
```

Comment:
Here and in some places above worth to use
```suggest
-url = "https://github.com/llvm/llvm-project/commit/1e8df9e85a1ff213e5868bd822877695f27504ad.patch";
+url = "https://github.com/llvm/llvm-project/commit/1e8df9e85a1ff213e5868bd822877695f27504ad.patch?full_index=1";
```

and update hashes appropriately. See #257446

Reply:
This is only required when using fetchpatch2, not fetchpatch as used here.

Reply:
@alyssais My colleague have sha256 mismatch downloading specifically this patch. Do you have any idea why?

Reply:
If your colleague has a hash mismatch, but you don't, then you could both diff your respective versions of the patch and figure out what's changing.  I can reproduce the patch with the hash present here.